src/app/frontend/replicasetdetail/sortedheader_controller.js

```diff
@@ -0,0 +1,69 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Controller for the sorted header directive.
+ * @final
+ */
+export default class SortedHeaderController {
+  constructor() {
+    /**
+     * Tooltip message for current header column.
+     * @export {string}
+     */
+    this.tooltip;
+
+    /**
+     * Name of current header column.
+     * @export {string}
+     */
+    this.key;
+
+    /**
+     * Name of column, that is currently used for sorting.
+     * @export {string}
+     */
+    this.sortKey;
+
+    /**
+     * If true then reverse sorting order will be applied.
+     * @export {boolean}
+     */
+    this.sortOrder;
+  }
+
+  /**
+   * Applies sorting on specific column of the table.
+   * @export
+   */
+  sort() {
+    this.sortOrder = (this.sortKey === this.key) ? !this.sortOrder : false;
+    this.sortKey = this.key;
+  }
+
+  /**
+   * Returns true if sorting on specific column and in given order of the table is applied.
+   * @param {boolean} sortOrder
+   * @return {boolean}
+   * @export
+   */
+  isSortedBy(sortOrder) { return this.sortKey === this.key && this.sortOrder === sortOrder; }
```

Comment:
You can test this file easily. Do not test html template, but only this controller. It does contain a little bit of logic, i.e., this method has this nice double if. You should test it :)